File: Java/Eta/Applications/Examples/com/thomsonreuters/upa/examples/common/DirectoryRequestInfo.java
package com.thomsonreuters.upa.examples.common;

import com.thomsonreuters.upa.transport.Channel;
import com.thomsonreuters.upa.valueadd.domainrep.rdm.directory.DirectoryMsgFactory;
import com.thomsonreuters.upa.valueadd.domainrep.rdm.directory.DirectoryMsgType;
import com.thomsonreuters.upa.valueadd.domainrep.rdm.directory.DirectoryRequest;

/**
 * Source directory request information.
 */
public class DirectoryRequestInfo
{
    Channel chnl;
    boolean isInUse;
    DirectoryRequest directoryRequest;
    
    public DirectoryRequestInfo()
    {
        directoryRequest = (DirectoryRequest) DirectoryMsgFactory.createMsg(); 
        directoryRequest.rdmMsgType(DirectoryMsgType.REQUEST);
    }
    
    public void clear()
    {
        chnl = null;
        isInUse = false;
        directoryRequest.clear();
    }

    public Channel channel()
    {
        return chnl;
    }
    
    public boolean isInUse()
    {
        return isInUse;
    }
    
    public DirectoryRequest directoryRequest()
    {
        return directoryRequest;
    }
}

File: CSharp/Eta/Applications/Examples/EtaAppCommon/Rdm/DirectoryRequestInfo.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2023-2024 LSEG. All rights reserved.
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.Transports;
using LSEG.Eta.ValueAdd.Rdm;

namespace LSEG.Eta.Example.Common
{
    public class DirectoryRequestInfo
    {
        public IChannel? Channel { get; internal set; }

        public bool IsInUse { get; internal set; }

        public DirectoryRequest DirectoryRequest { get; private set; } = new DirectoryRequest();

        public DirectoryRequestInfo()
        {
            Clear();
        }

        public void Clear()
        {
            Channel = null;
            IsInUse = false;
            DirectoryRequest.Clear();
        }
    }
}